### PR TITLE
[BLU-2832] Updates to deployment script

### PR DESCRIPTION
**Pull request by 0xmichalis**
Breaking the script back to two separate scripts where one just deploys the bridge and another one deploys everything to avoid mixing production with testing code which is an anti-pattern (I already had to fix the old version of the script [here](https://github.com/regen-network/toucan-bridge/pull/53/commits/777681ba82f8dd303a1e6965cad6fe6d2c078422) because both prod and testing code was running at the same time).

I have also made the token issuer optional as it is optional in the contract too and can be set retrospectively.

Also adding support for private keys, moving signer configuration from the deploy script in the Hardhat config.

Lastly updating the workflows to avoid running in draft PRs.

### CODE REVIEW

**Comment by 0xmichalis** [comment]
cc @aspiers

File: README.md [hunk 1]
```diff
@@ -47,28 +47,38 @@ git submodule update --remote --merge
...
 
-The bridge deploy script expects two environment variables. 1) address of the bridge controller: `BRIDGE_CONTROLLER_ADDRESS` and 2) address of the Toucan contract registry: `TOUCAN_CONTRACT_REGISTRY_ADDRESS`
+Copy `.env.example` to `.env` and set the desired configuration in `.env`.
 
 #### Testnets/Matic
 
 Before you deploy you should set up your signer wallet. You can either pass a mnemonic via the environment variable `MNEMONIC` or create a `mnemonic.txt` file with the seed phrase in the root of the repository.
 
+```shell
+yarn hardhat run scripts/deploy.js --network [matic|mumbai]
```
Comment (aspiers) [inline README.md:59]: Why don't we have the deployment code under `deploy/`?  I thought that was the expected place for it.

**Review by aspiers — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (0xmichalis) [reply README.md:59]: I think `scripts` is as expected or even more than `deploy` given that Hardhat docs use `scripts` for their examples[1]. Btw `deploy` is limiting as it excludes upgrade, or any other non-deploy related scripts. Given how we have structured `tokenizer`, `deploy` to me feels like a place to put deployment-related logic (not necessarily top-level) and `scripts` holds top-level scripts (that may or may not use code from `deploy`, given that as I understand it, `deploy` is standardized by `hardhat-deploy`).

[1] https://hardhat.org/hardhat-runner/docs/guides/deploying

Comment (aspiers) [reply README.md:59]: https://github.com/wighawag/hardhat-deploy#the-deploy-task says:

image

so we could configure it to be something else, but I don't see the benefit of doing that.  The intended way of selecting a subset of the deployment scripts to run is by using [`--tags`](https://github.com/wighawag/hardhat-deploy#deploy-scripts-tags-and-dependencies).  I'm not sure I follow your point about why putting deploy scripts in `deploy/` is limiting?  Upgrades could go elsewhere, and certainly non-deploy related scripts must not go in `deploy/`.

**Pull request merged**